### PR TITLE
Support DialContext on Pool

### PR DESCRIPTION
**Pull request by izumin5210**
#406

### CODE REVIEW

File: redis/pool.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import (
 	"sync"
 	"sync/atomic"
 	"time"
+
+	"golang.org/x/net/context"
```
Comment (izumin5210) [inline redis/pool.go]: above go 1.8, `golang.org/x/net/context` uses type alias to standard `context` package.
so it works well in all cases.

Comment (FZambia) [reply redis/pool.go]: Hi, does this mean extra dependency anyway?

Comment (izumin5210) [reply redis/pool.go]: it's an empty package for above go1.9 https://github.com/golang/net/blob/master/context/go19.go

**Comment by garyburd** [comment]
The new dependency is not acceptable.  It's OK to remove support for Go 1.6 and earlier.

**Comment by izumin5210** [comment]
> It's OK to remove support for Go 1.6 and earlier.

thanks 😄

**Comment by izumin5210** [comment]
i created a new PR to drop support for Go 1.6 and earier https://github.com/gomodule/redigo/pull/409

File: redis/pool.go (hunk 1 above)
Comment (izumin5210) [reply redis/pool.go]: removed on 21cf364

**Comment by garyburd** [comment]
The text description for the pool should say that Dial or DialContext must be set and that DialContext takes precedence over Dial when both are set.

**Comment by izumin5210** [comment]
thanks!!

**Pull request merged**